tasks/install.yml

```diff
@@ -1,38 +1,114 @@
+  when: previous_mattermost_version_stat.stat.exists
+
+- name: Set that mattermost needs to be updated
+  set_fact:
+    mattermost_needs_update:
+  when: not previous_mattermost_version_stat.stat.exists or previous_mattermost_version['content'] | b64decode != mattermost_version
+
+- block:
+  - name: 'Download team version binary from https://releases.mattermost.com'
+    get_url:
+      url: 'https://releases.mattermost.com/{{ mattermost_version }}/mattermost-team-{{ mattermost_version }}-linux-amd64.tar.gz'
+      dest: "{{ global_cache_dir | mandatory }}/"
+    when: not mattermost_enterprise
+    become: false
+    delegate_to: localhost
+    run_once: yes
+
+  - name: 'Download enterprise version binary from https://releases.mattermost.com'
+    get_url:
+      url: 'https://releases.mattermost.com/{{ mattermost_version }}/mattermost-{{ mattermost_version }}-linux-amd64.tar.gz'
+      dest: "{{ global_cache_dir | mandatory }}/"
+    when: mattermost_enterprise
+    become: false
+    delegate_to: localhost
+    run_once: yes
+
+  - name: Stop mattermost service
+    service:
+      name: mattermost
+      enabled: yes
+      state: stopped
+
+  - name: Clear a previously leaked temporary folder
+    file:
+      path: /opt/mattermost-displaced
+      state: absent
+
+  - name: Check if the mattermost folder exists
+    stat:
+      path: /opt/mattermost
+    register: previous_mattermost_folder_stat
+
+  - name: Clear the mattermost folder except for folders that are kept
+    block:
+      - command: mv /opt/mattermost /opt/mattermost-displaced
+
+      - file:
+          path: /opt/mattermost-displaced/{{ item }}
+          state: directory
+          recurse: yes
+        with_items: '{{ mattermost_preserve_on_update }}'
+
+      - debug:
+          msg: '{{ item_path | dirname }}'
+        vars:
+          item_path: /opt/mattermost/{{ item }}
+        with_items: '{{ mattermost_preserve_on_update }}'
+
+      - file:
+          path: '{{ item_path | dirname }}'
+          state: directory
+          recurse: yes
+        vars:
+          item_path: /opt/mattermost/{{ item }}
+        with_items: '{{ mattermost_preserve_on_update }}'
+
+      - command: mv /opt/mattermost-displaced/{{ item }} /opt/mattermost/{{ item }}
+        with_items: '{{ mattermost_preserve_on_update }}'
+
+      - file:
+          path: /opt/mattermost-displaced
+          state: absent
+
+    when: previous_mattermost_folder_stat.stat.exists
+
+  - name: Unpack Mattermost team archive
+    unarchive:
+      src: "{{ global_cache_dir }}/mattermost-team-{{ mattermost_version }}-linux-amd64.tar.gz"
+      dest: /opt/
+    args:
+    when: not mattermost_enterprise
+
+  - name: Unpack Mattermost enterprise archive
+    unarchive:
+      src: "{{ global_cache_dir }}/mattermost-{{ mattermost_version }}-linux-amd64.tar.gz"
+      dest: /opt/
+    args:
+    when: mattermost_enterprise
+
+  when: mattermost_needs_update is defined
```

Comment:
Please move these tasks to a new file e.g. `update_mattermost.yml` and include it with the condition